build/media-suite_helper.sh

```diff
@@ -2062,6 +2062,18 @@ check_signature() {
     gpg -q --verify "$sigFileName" "$@" > /dev/null 2>&1
 }
 
+do_jq() {
+    local jq_file="$jq_file"
+    # Detect if in pipe, useful for curling github api
+    if [[ ! -t 0 ]]; then
+        jq_file=/dev/stdin
+    elif [[ -f $1 ]]; then
+        jq_file="$1" && shift
+    fi
+
+    jq -r "$*" "$jq_file"
```

Comment:
Why `$*` instead of `$@`?

Reply:
jq does not support more than 1 json filter, anything after gets taken as a file parameter, so `curl ... | jq '.assets[]' '.'` is taken as wanting to open `.` as a json file

Reply:
Maybe just always redirect stdout to jq instead then, instead of using a file?

Reply:
We might want to use filtering/commands on jq, or certain options instead of always `-r`.

Reply:
> Maybe just always redirect stdout to jq instead then, instead of using a file?
That's the point of the if statements at the top

Reply:
I mean something like `< "$jq_file" jq -r -- "$@"`.

> jq does not support more than 1 json filter, anything after gets taken as a file parameter, so `curl ... | jq '.assets[]' '.'` is taken as wanting to open `.` as a json file

How is this an issue? It would also break `jq` itself. I don't think `"$@"` separates on spaces, so if you do `youtube-dl -J dQw4w9WgXcQ | jq -r '{title, album} | .title'` it wouldn't try to find a file called `album}`.
 
Speaking of which, `youtube-dl -J dQw4w9WgXcQ | jq -r '{title, album} | .title'` works with `jq` but not with `do_jq`.

Reply:
`< "$jq_file" jq -r "$@"` worked

Reply:
jq doesn't seem to have support for `--` parsing args stop

Reply:
> How is this an issue? It would also break `jq` itself. I don't think `"$@"` separates on spaces, so if you do `youtube-dl -J dQw4w9WgXcQ | jq -r '{title, album} | .title'` it wouldn't try to find a file called `album}`.

The `'{title, album} | .title'` would be taken as one due to the single quotes.

>@
Expands to the positional parameters, starting from one. When the expansion occurs within double-quotes, and where field splitting (see Field Splitting) is performed, each positional parameter shall expand as a separate field, with the provision that the expansion of the first parameter shall still be joined with the beginning part of the original word (assuming that the expanded parameter was embedded within a word), and the expansion of the last parameter shall still be joined with the last part of the original word. If there are no positional parameters, the expansion of '@' shall generate zero fields, even when '@' is double-quoted.

https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html

With `"$@"`, when running
```bash
youtube-dl -J dQw4w9WgXcQ | do_jq '{title, album} | .title' '|.'
```
`jq: error: Could not open file |.: Invalid argument`

With `$*`, when running 
```bash
youtube-dl -J dQw4w9WgXcQ | do_jq '{title, album} | .title'\
 '|.'
```
 `Rick Astley - Never Gonna Give You Up (Video)`

We can use newlines if it's `$*` without needing to keep the single quote open

Reply:
Yeah, but in what use case would you want `do_jq '{title, album} | .title' '|.'` to work?

Reply:
Ah, sorry, didn't read the last line.

Reply:
> ```shell
> youtube-dl -J dQw4w9WgXcQ | do_jq '{title, album} | .title\
>  |.'
> ```

Doesn't this work?

Reply:
Yes, it does

Reply:
> We might want to use filtering/commands on jq, or certain options instead of always `-r`.

Made it so that if there's an arg that starts with `-`, it will pass the options to jq as is `curl ... | do_jq -c .title` will run `jq -c .title` else it will run jq with `-r` on by default for easy bash parsing